src/battle_script_commands.c

```diff
@@ -4665,9 +4669,41 @@ static void Cmd_moveend(void)
             }
             gBattleScripting.moveendState++;
             break;
+        case MOVEEND_DANCER: // Special case because it's so annoying
+            if (gBattleMoves[gCurrentMove].flags & FLAG_DANCE)
+            {
+                u8 battler, nextDancer = 0;
+                
+                if (!(gBattleStruct->lastMoveFailed & gBitTable[gBattlerAttacker]
+                    || (gSpecialStatuses[gBattlerAttacker].dancerUsedMove == 0
+                        && gProtectStructs[gBattlerAttacker].usesBouncedMove)))
+                {   // Dance move succeeds
+                    // Set target for other Dancer mons; set bit so that mon cannot activate Dancer off of its own move
+                    if (gSpecialStatuses[gBattlerAttacker].dancerUsedMove == 0)
+                    {
+                        gBattleScripting.savedBattler = gBattlerTarget | 0x4;
```

Comment:
What's the deal with the bits here? Could you make them without orring and just use regular u8? Any reason you decided to perfrom them in the first place?

Reply:
If a mon uses a move like Revelation Dance against a particular target, its ally will have the same target. That's why the target chosen by the original mon has to be saved. It's ORRED so that battler 0 can be saved as well.